src/lighteval/models/nanotron/nanotron_model.py

```diff
@@ -509,25 +505,28 @@ def prepare_batch(
         if max_context is None:
             max_context = self.max_length
 
-        if max_context % self.parallel_config.tp != 0:
-            # We need to round up to the next multiple of self.parallel_config.tp
-            if (max_context + (self.parallel_config.tp - max_context % self.parallel_config.tp)) < self.max_length:
-                # We can add some tokens
-                max_context = max_context + (self.parallel_config.tp - max_context % self.parallel_config.tp)
-            else:
-                # We need to remove some tokens
-                max_context = max_context - (max_context % self.parallel_config.tp)
-
-        if padding_length % self.parallel_config.tp != 0:
-            # We need to round up to the next multiple of self.parallel_config.tp
-            if (
-                padding_length + (self.parallel_config.tp - padding_length % self.parallel_config.tp)
-            ) < self.max_length:
-                # We can add some tokens
-                padding_length = padding_length + (self.parallel_config.tp - padding_length % self.parallel_config.tp)
-            else:
-                # We need to remove some tokens
-                padding_length = padding_length - (padding_length % self.parallel_config.tp)
+        assert (
+            self.parallel_config.tp_mode == TensorParallelLinearMode.ALL_REDUCE
+        ), "No reason to have tp_mode==REDUCE_SCATTER when doing inference"
+        # if max_context % self.parallel_config.tp != 0:
+        #     # We need to round up to the next multiple of self.parallel_config.tp
+        #     if (max_context + (self.parallel_config.tp - max_context % self.parallel_config.tp)) < self.max_length:
+        #         # We can add some tokens
+        #         max_context = max_context + (self.parallel_config.tp - max_context % self.parallel_config.tp)
+        #     else:
+        #         # We need to remove some tokens
+        #         max_context = max_context - (max_context % self.parallel_config.tp)
+
+        # if padding_length % self.parallel_config.tp != 0:
+        #     # We need to round up to the next multiple of self.parallel_config.tp
+        #     if (
+        #         padding_length + (self.parallel_config.tp - padding_length % self.parallel_config.tp)
+        #     ) < self.max_length:
+        #         # We can add some tokens
+        #         padding_length = padding_length + (self.parallel_config.tp - padding_length % self.parallel_config.tp)
+        #     else:
+        #         # We need to remove some tokens
+        #         padding_length = padding_length - (padding_length % self.parallel_config.tp)
```

Comment:
rm

Reply:
cc @NouamaneTazi, do we want to keep this ?